lib/localenv/credentials/credentials.go

```diff
@@ -165,7 +187,7 @@ func (s *credentialsService) For(clusterURL string) (*Credentials, error) {
 		s.Debugf("Returning default credentials for %v.", clusterURL)
 		return defaultCredentials, nil
 	}
-	return nil, trace.NotFound("no credentials for %v", clusterURL)
+	return nil, trace.AccessDenied("no credentials for %v", clusterURL)
```

Comment:
Nit: I sort of feel like the original NotFound is a more appropriate error here. It's not a I tried to access something so I was denied access, I tried to locate a set of credentials in various locations and didn't find any??

Reply:
Yeah, the problem with returning NotFound errors everywhere is that it becomes impossible to distinguish between different cases which the caller has to act differently upon (e.g. "no currently active cluster" vs "no credentials for specified cluster") so I felt like access denied would fit here and also allow to separate the two scenarios.

Reply:
So my hesitation with something like this, is it creates a silent contract between the caller/callee, on what a NotFound / AccessDenied errors don't really mean AccessDenied/NotFound, but are returned in a specific way to convey some other context to the caller.

Where as I believe the current go idomatic pattern (not the go2 stuff), would be to have Is functions that allowed the caller to ask what type of error it got (like IsTimeout, IsPermission, IsExist in the os package) and then the caller can use that behavior accordingly, as an explicit contract between packages.

Alternatively, I added the ability to embed metadata in the errors, as custom fields, and an envisioned use case around this would be for embedding additional metadata that a caller could use, although we should likely settle on some conventions around that so we're consistent. But it's now possible to add kvp fields to trace errors, and read those from a caller to have the caller make choices.

As a very rough example, but ideally we would converge on a common pattern:
```
const ErrNoCurrentActiveCluster = "no_current_active_cluster"

func E() error {
     return trace.NotFound("no credentials found").AddFields({
        "cluster": clusterURL,
        ErrNoCurrentActiveCluster: true,
     })
}

func IsErrNoCurrentActiveCluster(err trace.TraceError) bool {
    _, ok := trace.GetFields(err)[ErrNoCurrentActiveCluster];
    return ok
}

// some other package
Func D() {
    err := package.E()
    if err != nil && package.IsErrNoCurrentActiveCluster(err) {
        // Do something
    }
}
```

Or we could go the other way, and make a convention for an error field of "sub_type", and embed the error as a string.